src/audio.rs

```diff
@@ -0,0 +1,86 @@
+use crate::twilio_response;
+
+pub struct BufferData {
+    pub inbound_buffer: Vec<u8>,
+    pub outbound_buffer: Vec<u8>,
+    pub inbound_last_timestamp: u32,
+    pub outbound_last_timestamp: u32,
+}
+
+/// if more than 400 ms of audio is available in both inbound and outbound
+/// audio buffers, we will drain that audio data, mix it together as correct
+/// stereo audio, taking into account any potentially dropped audio packets,
+/// and return the mixed audio
+pub fn process_twilio_media(
+    media: twilio_response::EventMedia,
+    mut buffer_data: &mut BufferData,
+) -> Option<Vec<u8>> {
+    // NOTE: when Twilio sends media data, it should send 20 ms audio chunks at a time, where each ms of audio is 8 bytes
+    let media_chunk = base64::decode(media.payload).unwrap();
+    let media_chunk_size = media_chunk.len();
+    if media_chunk_size != 20 * 8 {
+        // WARN: Twilio media chunk size is not the expected size of 20 * 8 bytes.
+    }
+    // NOTE: I've seen cases where the timestamp is less than 20 ms ahead of the previous chunk
+    let timestamp = media.timestamp.parse::<u32>().unwrap();
+
+    if media.track == "inbound" {
+        let time_lost = if timestamp < buffer_data.inbound_last_timestamp + 20 {
+            // WARN: Received inbound timestamp is less than 20 ms ahead of previous timestamp.
+            0
+        } else {
+            (timestamp - (buffer_data.inbound_last_timestamp + 20))
+                .try_into()
+                .unwrap()
+        };
+        if time_lost > 0 {
+            // silence for mulaw data is 0xff, and there are 8 bytes of mulaw data per ms
+            let silence = &vec![0xff; 8 * time_lost];
+            buffer_data.inbound_buffer.extend_from_slice(&silence[..]);
+        }
+        buffer_data
+            .inbound_buffer
+            .extend_from_slice(&media_chunk[..]);
```

Comment:
Slightly shorter but :shrug: 
```suggestion
        buffer_data.inbound_buffer.extend(media_chunk);
```